propagation/b3/spancontext.go

```diff
@@ -88,3 +88,113 @@ func ParseHeaders(
 
 	return sc, nil
 }
+
+// ParseSingleHeader takes values found from B3 Single Header and tries to reconstruct a
+// SpanContext.
+func ParseSingleHeader(contextHeader string) (*model.SpanContext, error) {
```

Comment:
I find it hard to read all those positional slice arguments...

I would personally do a split on "-" and have a code flow where the amount of items resulting from this split dictate what kind of b3 single header we got and have all items clearly named for their positional arguments.

e.g.  single item can only mean we expect a sample flag: 0, 1, d
two items means we only have traceID + spanID
three items means we have traceID + spanID + debug
four items means we also have parentSpanID (and sample flag could be empty (meaning undecided))

Reply:
another option could be to define constants for the offsets so by naming it, it is more clear what the payload of the resulting slices is.